### PR TITLE
Upgrade kafka playbooks now make sure controller is last to be touched before any restarts

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Now the upgrade kafka playbooks take extra care to do restarts on the controller last

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Tested locally, with 3 broker cluster, will run it through jenkins as well

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tasks/create_ordered_kafka_groups.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+# Run these tasks over kafka_broker group with serial: 1
+# Creates two groups: controller and non_controllers
```
Comment (domenicbove) [inline tasks/create_ordered_kafka_groups.yml]: comment is incorrect

File: upgrade_kafka_broker.yml [hunk 2]
```diff
@@ -41,13 +48,19 @@
         line: "inter.broker.protocol.version={{inter_broker_protocol_version}}"
         regexp: inter.broker.protocol.version.*
         state: present
+      # notify:
+      #   - Restart Kafka
```
Comment (domenicbove) [inline upgrade_kafka_broker.yml]: @JumaX Curious your take on these kafka properties/ if a restart is necessary. The scenario is this: kafka is at version 2.3, and log.message.format.version is not set. The playbook will set it to 2.3, but wont restart kafka (bc its not necessary). Then later kafka gets upgraded to 2.4 and on that restart its important the version stays at 2.3.

File: upgrade_kafka_broker.yml [hunk 3]
```diff
@@ -155,7 +159,7 @@
       import_tasks: tasks/wait_for_urp.yml
 
 - name: Kafka Broker Upgrade - Set Inter-Broker Protocol to Current Version
-  hosts: kafka_broker
+  hosts: kafka_broker_non_controllers,kafka_broker_controller
```
Comment (domenicbove) [inline upgrade_kafka_broker.yml]: there needs to be another call to the ordering tasks bc do not know the controller is the same

File: upgrade_kafka_broker.yml (hunk 2 above)
Comment (JumaX) [reply upgrade_kafka_broker.yml]: @domenicbove so if I understand, we are simply writing the property so we have it for the next upgrade, as the Broker is already set to that properties value.  In this case, not restarting makes sense.

File: upgrade_kafka_broker.yml [hunk 4]
```diff
@@ -41,13 +48,19 @@
...
+      #   - Restart Kafka
+      #   - Wait for URP
 
     - name: Set Log Message Format Version to Current Version
       lineinfile:
         name: "{{kafka_broker.config_file}}"
         line: "log.message.format.version={{log_message_format_version}}"
         regexp: log.message.format.version.*
         state: present
+      # notify:
```
Comment (JumaX) [inline upgrade_kafka_broker.yml]: Remove comments.

File: upgrade_kafka_broker.yml [hunk 5]
```diff
@@ -41,13 +48,19 @@
         line: "inter.broker.protocol.version={{inter_broker_protocol_version}}"
         regexp: inter.broker.protocol.version.*
         state: present
+      # notify:
```
Comment (JumaX) [inline upgrade_kafka_broker.yml]: Remove comments.

**Pull request merged**